tf/tfprocess.py

```diff
@@ -159,10 +159,15 @@ def __init__(self, cfg):
         self.renorm_max_d = self.cfg['training'].get('renorm_max_d', 0)
         self.renorm_momentum = self.cfg['training'].get(
             'renorm_momentum', 0.99)
-
+   
         if self.cfg['gpu'] == 'all':
             self.strategy = tf.distribute.MirroredStrategy()
             tf.distribute.experimental_set_strategy(self.strategy)
+        elif self.cfg['gpu'].__contains__(','):
+            gpus = tf.config.experimental.list_physical_devices('GPU')
+            for i in self.cfg['gpu'].split(","):
+                tf.config.experimental.set_visible_devices(gpus[int(i)],
```

Comment:
This doesn't give you multiple gpus active - you need strategy for that.  What you want to do I think is make a list and add gpus[int(i)] for each i.  Then pass that to tf.distribute.MirroredStrategy initializer after the loop - then rest of the code is same as the 'all' case.

Reply:
ok sure

Reply:
ohh i thought I added that
sorry didn't see